dash/backend/src/modules/pod/dao/pod.dao.ts

```diff
@@ -339,21 +339,24 @@ export class PodDao {
     async linkExistingPodImages(podId: number, imageIds: Array<number>): Promise<void> {
       const knex = await this.databaseService.getConnection();
 
-      const links = imageIds.map(id => ([podId, id]));
-
-      let existingLinks = await knex('pod_images').whereIn(['pod_id', 'image_id'], links) as [{pod_id: number, image_id: number}];
+      const links = imageIds.map(id => ({pod_id: podId, image_id: id}));
 
-      let newLinks: {pod_id: number,  image_id: number}[] = [];
+      await knex.transaction(async function(trx){
+        try {
+          // Clear all existing links to the pod
+          await knex.delete()
```

Comment:
Is there a history table these should go into or no?

Reply:
This is my understanding, let me know if this is incorrect or if you think we should confirm with Jake:
The sync job this is part of runs (by default) every 30minutes, and makes syncs all of the kubernetes data that is not part of the history.

The history job runs (by default) daily at 12:01, and saves as a point in time what is currently exists fromt he sync jobs.